sormas-backend/src/main/java/de/symeda/sormas/backend/deletionconfiguration/AbstractCoreEntityFacade.java

```diff
@@ -38,9 +44,15 @@ public void executeAutomaticDeletion(DeletionConfiguration entityConfig) {
 
 		List<T> toDeleteEntities = QueryHelper.getResultList(em, cq, null, null);
 
-		toDeleteEntities.forEach(entity -> {
-			delete(entity);
-		});
+		toDeleteEntities.forEach(this::delete);
+	}
+
+	public AutomaticDeletionInfoDto getAutomaticDeletionInfo(String uuid, CoreEntityType coreEntityType) {
+		DeletionConfiguration deletionConfiguration = deletionConfigurationService.getCoreEntityTypeConfig(coreEntityType);
+		Object[] deletionData = getDeletionData(uuid, deletionConfiguration);
+		Date referenceDate = (Date) deletionData[0];
```

Comment:
Have you tested this with missing deletion references and periods? Pretty sure that you will run into all sorts of null pointers in this method and in `getDeletionData` below (e.g. in `addDays` when the deletionPeriod is null, or in `getDeleteReferenceField` when deletionReference is null).